tests/unit/internal-session-test.js

```diff
@@ -189,6 +196,38 @@ describe('InternalSession', () => {
     describe('when the restored data does not contain an authenticator factory', () => {
       itDoesNotRestore();
     });
+
+    describe('with older synchronous stores (< v1.1.0)', function() {
+      describe('when restoring the session', function() {
+        describe('when the store resolves restoration', function() {
+          beforeEach(() => {
+            sinon.stub(store, 'restore').returns({ authenticated: { authenticator: 'authenticator' } });
+            sinon.stub(store, 'persist').returns();
+            sinon.stub(store, 'clear').returns();
+          });
+
+          it('is authenticated', () => {
+            return session.restore().then(() => {
+              expect(session.get('isAuthenticated')).to.be.true;
+            });
+          });
+        });
+
+        describe('when the store rejects restoration', function() {
```

Comment:
this case should be tested for authentication and invalidation as well (when the store rejects in `_setup` or `_clear`). Especially when it rejects in `_clear` the session must remain authenticated.